.ci/run-link-check-plugin.sh

```diff
@@ -7,8 +7,34 @@ pwd
 uname -a
 mvn --version
 curl --fail-with-body -I https://sourceforge.net/projects/checkstyle/
-mvn -e --no-transfer-progress clean site -Dcheckstyle.ant.skip=true -DskipTests -DskipITs \
-   -Dpmd.skip=true -Dspotbugs.skip=true -Djacoco.skip=true -Dcheckstyle.skip=true
+
+# Function to calculate the replacement string based on the file depth
+calculate_replacement() {
+    local file_path="$1"
+    local relative_path="${file_path#src/main/java/}"
+    local depth=$(echo "$relative_path" | grep -o "/" | wc -l)
+    local replacement=""
+    for ((i=0; i<depth; i++)); do
+        if ((i > 0)); then
+            replacement="/$replacement"
+        fi
+        replacement="..$replacement"
+    done
+    echo "../$replacement"
+}
+
+# convert all javadoc website links to relative links
+grep -rl "https://checkstyle.org" src/main/java | while read -r file; do
+    replacement=$(calculate_replacement "$file")
+    sed -i "s|https://checkstyle.org|$replacement|g" "$file"
+done
+
+mvn -e --no-transfer-progress -Pno-validations clean site -Dlinkcheck.skip=false \
+   -Dmaven.javadoc.skip=false
+
+# revert all modified javadoc files for post validation
+git checkout HEAD -- src/main/java
```

Comment:
this is a concern, I usually always treated CI tasks or any validation task as commands that ready to be executed at any time in root of repo.
this update will revert all user uncommited changes, not good.
if we agreed to move forward with this links changes we need to copy checkouted repo to .ci-temp and do all code changes in that folder, and this checkout might not be required at all.